### DISCUSSION TITLE
Cannot compile gcc

### DISCUSSION THREAD

**Discussion by zap8600** [termux/termux-packages#10708] [General]
### Problem description

I'm trying to compile gcc-12.1.0. At a certain point it says that it needs to be a position independent executable. I've added `-fPIE` to `LDFLAGS`, but that didn't fix it. 

### Steps to reproduce the behavior.

Compile gcc.

### System information

* Termux application version: v0.118.0
* Android OS version: 10
* Device model: SM-G965U

**Comment by Grimler91** [comment]
You haven't mentioned how you are building it, or which configure args you have used, so impossible for anyone to reproduce your issue.

Have you tried adding -fPIE to CPPFLAGS, or CFLAGS instead?

**Reply by zap8600** [reply]
Here is how my gcc is currently being configured. 
`CFLAGS+="-fPIE" CPPFLAGS+="-fPIE" ../gcc-12.1.0/configure --target=i686-elf --prefix="$HOME/opt/cross" --disable-nls --enable-languages=c,c++ --without-headers`
As you can tell, it didn't work.

**Comment by xtkoba** [comment]
I would suggest adding `gcc_cv_c_no_fpie=no` and `gcc_cv_no_pie=no` as arguments for `./configure`. Or patch `gcc/configure{,.ac}` to drop those flags.

**Reply by zap8600** [reply]
Adding `gcc_cv_c_no_fpie=no` and `gcc_cv_no_fpie=no` didn't work. I don't know what to patch in configure and configure.ac.

**Reply by xtkoba** [reply]
Like this:

https://github.com/ibara/misc/blob/b10af089be9db1d5064d0d71c184d7d425df3680/gnu-d-compiler/gdc-openbsd.diff#L76-L113

**Reply by zap8600** [reply]
Even though I patched it, it still didn't work.

**Reply by zap8600** [reply]
If I patch configure, do I need to add `gcc_cv_c_no_fpie` or not?

**Reply by xtkoba** [reply]
Should not be needed if patched.

Also `--enable-default-pie` as a configure arg, maybe. I would `grep -i fno-pie` some logs (terminal output and `config.log`s) and see what's going on.

**Reply by zap8600** [reply]
So gcc compiled successfully. When I tried to compile [my OS](https://github.com/zap8600/CCOS), i686-elf-ld gave me an error saying `undefined reference to _GLOBAL_OFFSET_TABLE_`.
